src/webui.c

```diff
@@ -175,10 +197,19 @@ process_request(struct evhttp_request *req, void *arg)
         return;
     }
 
+    if (strstr(url, "/worker/") != NULL)
+    {
+        char rig_id[MAX_RIG_ID];
+        char *uri_buffer = strstr(url, "/worker/");
+        uri_buffer += 8;
+        strncpy(rig_id, uri_buffer, MAX_RIG_ID);
+        send_json_worker(req, arg, rig_id);
```

Comment:
This needs refactoring. 1) you've already matched /worker/ above, 2) as you're pulling the worker name from the URL, you need to URL decode (i.e. evhttp_uridecode).

Reply:
the above should match /workers

Reply:
You're already doing the search on [line 200](https://github.com/jtgrassie/monero-pool/pull/82/files/bbed4c1a5737283d6d216668728ea8d14539d1bf#diff-5a767194e237ae8537db17efeb4d35abadd8071cacd2b600baa3dfd2a33acba4R200), so just store the result there to avoid the 2nd same strstr search.